### PR TITLE
[skip-ci][tree] Mark TEntryList and TEventList legacy

### LINKED ISSUE #14544: [ROOT-8515] Make TEntryList class reference relevant
### Explain what you would like to see improved and how.

From https://its.cern.ch/jira/browse/ROOT-8515

https://root.cern.ch/doc/master/classTEntryList.html is dominated by two large, ugly pieces of old code, that have nothing to do with TEntryList but generate graphics. I would even argue that those graphics don't help in the understanding of TEntryList and can be removed.
Additionally, both TEntryList and TEventList should inform the reader about which one is deprecated and which one to use. They should do this for the deprecated version right as the first sentence of the doc, so people don't waste time reading up on an irrelevant interface.
Thanks!
Axel.

### ROOT version

Any

### Installation method

Any

### Operating system

Any

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dpiparo**
and reorganise the documentation of TEntryList.

Fixes #14544

### CODE REVIEW

**Comment by pcanal** [comment]
> Mark TEntryList and TEventList legacy

Do you mean

> Mark TEventList legacy and make TEntryList documentation more prominent

?

**Comment by dpiparo** [comment]
Thanks for the comment. I adopted a more aggressive posture on purpose. Is TEntryList something we want to further develop and encourage to use?

File: tree/tree/src/TEntryList.cxx [hunk 1]
```diff
@@ -12,38 +12,9 @@
 /** \class TEntryList
 \ingroup tree
 
-A List of entry numbers in a TTree or TChain.
-
-There are two types of entry lists:
+\legacy{TEventList}
```
Comment (ferdymercury) [inline tree/tree/src/TEntryList.cxx:15]:
```suggestion
\legacy{TEntryList}
```

Comment (pcanal) [reply tree/tree/src/TEntryList.cxx:15]: `TEntryList` is the (more) modern interface and should not be marked legacy for now.

**Comment by pcanal** [comment]
> Is TEntryList something we want to further develop and encourage to use?

It is at the same level of usage and encouragement as `TTree`.  (And if I remember correctly we do not have yet a `RNTuple` replacement/equivalent).

**Comment by vepadulano** [comment]
> It is at the same level of usage and encouragement as TTree. 

I sincerely hope not. TEntryList is definitely not usable in production contexts as it only scales well for a few entries for its purpose. For RDataFrame we had to move away from it exactly for this reason. 

> And if I remember correctly we do not have yet a RNTuple replacement/equivalent).

Indeed, and I hope we are not going to develop the same feature.

IMHO, for the purpose of `TEntryList`, we have already `TTreeReader::SetEntryRange` and we should have similar interfaces (private/internal) for modern I/O systems.

**Comment by dpiparo** [comment]
Hi. Included all the comments (thanks to the 3 of you!) in a new version of the PR: I hope this is a good balance!

**Review by ferdymercury — APPROVED** [review]

**Comment by pcanal** [comment]
> IMHO, for the purpose of TEntryList, we have already TTreeReader::SetEntryRange

`SetEntryRange` is small subset of the feature/use of `TEntryList`.   The main intent of the `TEntryList` is to store the result of a selection (so a more or sparse list entry numbers) so that further analysis step can read only the already select entries.

> TEntryList is definitely not usable in production contexts as it only scales well for a few entries for its purpose

I am not sure I understand 'why'.  For `TTree::Draw`, this is not the case as far as I remember.  Does `TEventList` scale better for the case you are talking about?  (if that is the case we would need to NOT deprecated either of them).

Side note: the main intent of `TEntryList` over `TEventList` is that `TEntryList` is supposed to scale better with chain with large number of files.

> Indeed, and I hope we are not going to develop the same feature.

Humm ... it does not have to be implemented in `RNTuple` and could be implemented in `RDataFrame`
but why don't we want to have the ability to 'store' the result of computational heavy selection to speed-up further analysis by reducing the number of entries needed to be used?

**Pull request merged**